[Tan Yu Meng] iP

yumengtan opened the pull request:
# **Lucifer** 😈

> Just think of me as your hot, sexy, irresistible guardian devil. - Lucifer, probably

Are you down bad like [these people](https://www.youtube.com/watch?v=fbVRnlR_6BM&ab_channel=BuzzFeedCeleb)?
Fret not! Your favourite devil is now here to serve you miscreants!
Fantasize it as your virtual Tom Ellis that assists you with your desires. It's:

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/yumengtan/ip/archive/refs/tags/A-Jar.zip).
2. double-click it.
3. add your desires.
4. let it manage your desires for you 👅

And it is **FREE!**

Features: 
- [x] Managing desires
- [x] Managing deadlines
- [ ] Reminders (coming soon)
- [ ] And much more (coming soon)

You can use it to practice Java programming too. Here's the main method:
```Java
public static void main(String[] args) throws IOException {
        new Lucifer().run();
    }
```

Commit:
Add Gradle support

yumengtan committed:
level 1

yumengtan committed:
Level 2. Add, List

yumengtan committed:
Level 3. Mark as Done

yumengtan committed:
Level 4. ToDos, Events, Deadlines

yumengtan committed:
one word command error

yumengtan committed:
A-TextUITesting: Automated Text UI Testing

yumengtan committed:
Level 5. Handle Errors.

yumengtan committed:
Handled Upper case for commands

yumengtan committed:
Level 6. Delete

yumengtan committed:
Added Exceptions to package

yumengtan committed:
added help commands

yumengtan committed:
fixed todo/deadline/event command not working

yumengtan committed:
Save

yumengtan committed:
Fixed Load function

yumengtan committed:
Add Date and Time

yumengtan committed:
Merge tag 'Level-7'

# Conflicts:
#	src/main/java/Lucifer.java
#	src/main/java/Storage.java

yumengtan committed:
Fixed loadlist error

yumengtan committed:
Add A-MoreOOP

yumengtan committed:
Add A-Packages

yumengtan committed:
A-JUnit

yumengtan committed:
Change UI for empty list

yumengtan committed:
Add  A-Jar

yumengtan committed:
Add JavaDoc

yumengtan committed:
Add A-CodingStandard

yumengtan committed:
Add Find function

yumengtan committed:
Reformat functions

yumengtan committed:
Edit find function

Geralddtan commented on src/main/java/Commands/AddCommands.java:
```diff
@@ -0,0 +1,77 @@
+ * Class AddCommmands which inherits from ParseCommands.
+ * @author Yu Meng
+ */
+public class AddCommands extends ParseCommands{
```
Perhaps there should be a " " separator before the {. I have also noticed the same issue in some other places, notably in the AddCommandsTest.java file!

Geralddtan commented on src/main/java/Commands/AddCommands.java:
```diff
@@ -0,0 +1,77 @@
+     * @param desc the description of task
+     * @param old the user input format of the event date
+     */
+    public void addEvent(String desc, LocalDateTime old) {
```
Perhaps a more intuitive variable name here instead of "old"? This might give increased clarity to the reader!

Geralddtan commented on src/main/java/Commands/AddCommands.java:
```diff
@@ -0,0 +1,77 @@
+     * @param by the date and time of the task deadline
+     */
+    public void addDeadline(String description, LocalDateTime by) {
+        System.out.println(UNDERSCORE);
+        System.out.println("Got it. I have added this to your desires:");
+        Deadline curr = new Deadline(description, by);
+        list.add(curr);
+        System.out.println(curr);
+        System.out.println("Currently you have " + list.size() + " things yet to be desired");
+        System.out.println(UNDERSCORE);
```
Perhaps an abstraction method which performs the task of printing the UNDERSCORE lines might be a good addition! I've noticed several cases of this being repeated and abstracting this out to a separate method/class could be better!

Geralddtan commented on src/main/java/Commands/ParseCommands.java:
```diff
@@ -0,0 +1,129 @@
+    /**
+     * Parses the command from input.
+     */
+    public void parseCommand(String command) {
```
This method seems a little lengthy and is deeply nested, making it a little difficult to understand the proper flow. Perhaps a more concise method/abstraction methods could be employed!

Geralddtan commented on src/main/java/Lucifer.java:
```diff
@@ -0,0 +1,69 @@
+ * Main class which runs the Lucifer chatbot
+ * @author Yu Meng
+ */
+ public class Lucifer {
```
I really like the idea of using Lucifer :) Gives your task handler a great persona!

Geralddtan reviewed (commented):
Overall, I think your code structure is well organised! However, some methods seem pretty length and deeply nested, making it hard to decipher the logical flow. You also have followed the coding style practices quite closely with just a few issues pointed out such as having more informative variable names! Overall, LGTM, just a few nits to fix! Love the idea of using Lucifer!

yumengtan committed:
Edit for Coding Standard

yumengtan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

yumengtan committed:
Add A-CheckStyle

yumengtan committed:
Change Exception

yumengtan committed:
Add basic GUI

yumengtan committed:
Refactor packages

yumengtan committed:
Refactor packages and update GUI

yumengtan committed:
ParseCommand:
Implement Single Level of Abstraction Principle (SLAP) for better readability and avoid varying levels of abstraction.

Method previously had arrow-head style code due to many nesting implementations.

Edited method to adhere to Coding Standard and reduce nesting in method.

Lucifer:
Remove redundant methods and variables and set background for GUI.

DialogBox:
Improve UI of chatbox and profile picture.

AddCommandsTest:
Edit test methods based on changes in Lucifer class.

yumengtan committed:
Merge pull request #1 from yumengtan/A-CodeQuality

ParseCommand:

yumengtan committed:
Deadline:
Add assertion
Event:
Add assertion
Task:
Add assertion

yumengtan committed:
Merge pull request #2 from yumengtan/A-Assertions

Deadline:

yumengtan committed:
Add extension

yumengtan committed:
Merge branch 'master' of https://github.com/yumengtan/ip

yumengtan committed:
Update and improve GUI

yumengtan committed:
Change packages name to fit Coding Standard and change assert location

yumengtan committed:
Add user guide

yumengtan committed:
Set theme jekyll-theme-minimal

yumengtan committed:
Set theme jekyll-theme-cayman

yumengtan committed:
Set theme jekyll-theme-cayman

yumengtan committed:
Change output messages for methods

yumengtan committed:
Merge branch 'master' of https://github.com/yumengtan/ip

yumengtan committed:
Change save file path

yumengtan committed:
rename wrong name format for Ui.png

yumengtan committed:
Change wrong output message for adding tasks

yumengtan committed:
Edit README to remove markdown syntax

yumengtan committed:
Edit header comment to adhere to coding standard
